crates/sui-network/src/randomness/mod.rs

```diff
@@ -394,10 +398,19 @@ impl RandomnessEventLoop {
                 }) {
                 Ok(partial_sigs) => partial_sigs,
                 Err(e) => {
-                    debug!("failed to deserialize partial sigs: {e:?}");
+                    warn!("failed to deserialize partial sigs: {e:?}");
                     return;
                 }
             };
+        // Verify we received the expected share IDs (to protect against a validator that sends
+        // valid signatures of other peers which will be successfully verified below).
+        let received_share_ids = partial_sigs.iter().map(|s| s.index).collect::<Vec<_>>();
+        if received_share_ids != *expected_share_ids {
```

Comment:
can you avoid having to instantiate a new Vec by using `eq` instead of `collect`?